chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -160,14 +160,14 @@ impl EthereumAdapter {
         }
     }
 
-    fn traces(
-        &self,
-        logger: &Logger,
+    async fn traces(
+        self,
```

Comment:
Does this really need to be owned and then dropped? Seems like we should be able to keep `&self` here.